src/drivers/driver.ts

```diff
@@ -207,23 +207,24 @@ export abstract class CMakeDriver implements vscode.Disposable {
    * Get the environment variables that should be set at CMake-build time.
    */
   async getCMakeBuildCommandEnvironment(in_env: proc.EnvironmentVariables): Promise<proc.EnvironmentVariables> {
+    let envs: proc.EnvironmentVariables;
     if (this.useCMakePresets) {
-      return this._buildPreset?.environment as proc.EnvironmentVariables;
+      envs =  util.mergeEnvironment(in_env, this._buildPreset?.environment as proc.EnvironmentVariables);
```

Comment:
Should we revert this in case we decide to use Config Preset Environment variables using `getConfigureEnvironment `? Or do we still want the `in_env `to be used?

Reply:
I believe we want still in_env.

Reply:
@xisui-MSFT, maybe you want to confirm whether we should or shouldn't use in_env in getCMakeBuildCommandEnvironment here? Originally, before presets, in_env was used. With the support of presets, I don't know if you intentionally skipped merging with the incoming in_env or simply missed this doing a merge.

Reply:
This is intentional. All other sources of env are ignored when using presets.

Reply:
Ok, but the functionality fixed before presets in getCMakeBuildCommandEnvironment will not match when presets are on. I understand the difference between getCMake[Configure][BuildCommand][Test]Environment family but talking only about getCMakeBuildCommandEnvironment, with presets on and off, I am not sure about the difference.

Reply:
I think matching between presets on/off had never been the goal here, but we are trying more to match with the behavior of `cmake --preset` command when presets are on. And so this is why we ignore a large number of settings and defaults, including but not limited to environments, in pre-presets era.

Reply:
Makes sense, thank you @xisui-MSFT for your input.